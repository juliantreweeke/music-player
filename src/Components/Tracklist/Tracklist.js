import React from "react";
import styled from 'styled-components/macro'
import { Transition } from 'react-spring/renderprops';

const listStyles = `
  color:#000000;
  font-size:1em;
  list-style:none;
  overflow:hidden;
  margin:12px;
  max-width:400px;
`
const selectedStyles = `color:#FFFFFF; background:black`

const HeadingContainer =  styled.div`
  align-items:center;
  color:black;
  display:inline-flex;
  margin-top:24px;
  margin-left:24px;
  & h2 {
    margin-top:0px;
  }
  & p {
    margin-left:8px;
    margin-top:0px;
  }
`
const TrackContainer = styled.div`
  overflow:auto;
  height: 400px;
`

export default function Tracklist({querySearched, selectTrack, selectedTrack, tracks}) {
  return (
    <div>
        {querySearched && 
        <HeadingContainer>
          <h2>{querySearched}</h2>
          <p>{tracks.length} Search Results</p>
        </HeadingContainer>
          }
          <TrackContainer>
            <Transition
              items={tracks}
              keys={item => item}
              from={{ opacity: 0, height: 0 }}
              enter={{ opacity: 1, height: 75 }}
              leave={{ opacity: 0, height: 0 }}
            >
              {(item, state, index) => props => {
                const isTrackSelected = index === selectedTrack;
                return (
                  <li css={`${listStyles};${isTrackSelected && selectedStyles};`} style={{ ...props }}>
                    <p index={index} onClick={() => selectTrack(index)}>{index + 1} {item}</p>
                  </li>
              )}}
            </Transition>
          </TrackContainer>
    </div>
  );
};






