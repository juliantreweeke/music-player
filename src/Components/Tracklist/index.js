import React, { Component } from "react";
import Tracklist from './Tracklist.js'; 
import { connect } from 'react-redux';
import { actions } from '../../redux/ducks/index.js';

const mapStateToProps = state => ({
  ...state
 })

 const mapDispatchToProps = dispatch => ({
  resetTracks: () => dispatch(actions.resetTracks()),
  setData: (data) => dispatch(actions.setData(data)),
  setPlay: () => dispatch(actions.setPlay()),
  setTracks: (tracks) => dispatch(actions.setTracks(tracks)),
 })

class TracklistContainer extends Component {
  
  animateTitles = () => {
    const { data } = this.props;
    let timeOutLength = 100;
    if (data) {
      data.forEach(track => {
        setTimeout(() => this.props.setTracks([...this.props.tracks, track.title]), timeOutLength);
        timeOutLength += 200;
      })
    }
  }

   selectTrack = (i) => {
    let trackOrder = this.props.data;
    let temp = trackOrder[0];
    trackOrder[0] = trackOrder[i];
    trackOrder[i] = temp;
    this.props.setPlay();
    this.props.setData(trackOrder);
  }

  componentDidMount() {
    this.props.resetTracks();
    this.animateTitles();
  }
  render() {
      return <Tracklist
                tracks={this.props.tracks}
                selectTrack={this.selectTrack}
            />;
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TracklistContainer);