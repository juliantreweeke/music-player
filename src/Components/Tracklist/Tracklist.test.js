import React from 'react';
import { Tracklist } from './Tracklist';
import { cleanup, render} from '@testing-library/react'
import { store } from '../../redux/store/index.js';
import { Provider } from 'react-redux';

const mockTracks = [{ title: 'song1', index: 0 }, { title: 'song2', index: 1 }]
const querySearched = "Song"

describe('TrackList', () => {
    afterEach(cleanup)     
    
    it("should render", () => {   
         const TracklistComponent = render(
            <Provider store={store}>
                <Tracklist querySearched={querySearched} tracks={mockTracks} selectedTrack={0} />
            </Provider>)
          expect(TracklistComponent).toBeTruthy();  
    });

    it("matches snapshot", () => {
        const TracklistComponent = render(
            <Provider store={store}>
                <Tracklist querySearched={querySearched} tracks={mockTracks} selectedTrack={0} />
            </Provider>)
            expect(TracklistComponent).toMatchSnapshot();
    });

    it("matches snapshot with non default selectedTrack", () => {
        const TracklistComponent = render(
            <Provider store={store}>
                <Tracklist querySearched={querySearched} tracks={mockTracks} selectedTrack={1} />
            </Provider>)
            expect(TracklistComponent).toMatchSnapshot();
    });

    it("matches snapshot with no querySearched", () => {
        const TracklistComponent = render(
            <Provider store={store}>
                <Tracklist tracks={mockTracks} selectedTrack={1} />
            </Provider>)
            expect(TracklistComponent).toMatchSnapshot();
    });

    it("should have text with correct length of tracks", () => {  
        const { getByText } = render(
           <Provider store={store}>
               <Tracklist querySearched={querySearched} tracks={mockTracks} selectedTrack={0} />
           </Provider>)
           expect(getByText('2 Search Results')).toBeInTheDocument()
   });

   it("heading should be querySearched ", () => {  
    const { getByRole } = render(
       <Provider store={store}>
           <Tracklist querySearched={querySearched} tracks={mockTracks} selectedTrack={0} />
       </Provider>)
        expect(getByRole('heading')).toHaveTextContent(querySearched);
    });
});
