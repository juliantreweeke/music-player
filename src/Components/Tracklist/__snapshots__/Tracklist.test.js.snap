// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Tracklist__Card-sc-1odr728-0 hULKon"
      >
        <div
          class="Tracklist__HeadingContainer-sc-1odr728-1 jjKLpr"
        >
          <span>
            <h2>
              Song
            </h2>
            <p>
              2
               Search Results
            </p>
          </span>
        </div>
        <ul
          class="Tracklist__TrackContainer-sc-1odr728-2 fvBEZw"
        >
          <li
            class="Tracklist___StyledLi-sc-1odr728-3 cPRJbD"
            style="opacity: 0; height: 0px;"
          >
            <p>
              song1
            </p>
          </li>
          <li
            class="Tracklist___StyledLi-sc-1odr728-3 lnaxlV"
            style="opacity: 0; height: 0px;"
          >
            <p>
              song2
            </p>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Tracklist__Card-sc-1odr728-0 hULKon"
    >
      <div
        class="Tracklist__HeadingContainer-sc-1odr728-1 jjKLpr"
      >
        <span>
          <h2>
            Song
          </h2>
          <p>
            2
             Search Results
          </p>
        </span>
      </div>
      <ul
        class="Tracklist__TrackContainer-sc-1odr728-2 fvBEZw"
      >
        <li
          class="Tracklist___StyledLi-sc-1odr728-3 cPRJbD"
          style="opacity: 0; height: 0px;"
        >
          <p>
            song1
          </p>
        </li>
        <li
          class="Tracklist___StyledLi-sc-1odr728-3 lnaxlV"
          style="opacity: 0; height: 0px;"
        >
          <p>
            song2
          </p>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TrackList matches snapshot with no querySearched 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Tracklist__Card-sc-1odr728-0 hULKon"
      >
        <ul
          class="Tracklist__TrackContainer-sc-1odr728-2 fvBEZw"
        >
          <li
            class="Tracklist___StyledLi-sc-1odr728-3 lnaxlV"
            style="opacity: 0; height: 0px;"
          >
            <p>
              song1
            </p>
          </li>
          <li
            class="Tracklist___StyledLi-sc-1odr728-3 cPRJbD"
            style="opacity: 0; height: 0px;"
          >
            <p>
              song2
            </p>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Tracklist__Card-sc-1odr728-0 hULKon"
    >
      <ul
        class="Tracklist__TrackContainer-sc-1odr728-2 fvBEZw"
      >
        <li
          class="Tracklist___StyledLi-sc-1odr728-3 lnaxlV"
          style="opacity: 0; height: 0px;"
        >
          <p>
            song1
          </p>
        </li>
        <li
          class="Tracklist___StyledLi-sc-1odr728-3 cPRJbD"
          style="opacity: 0; height: 0px;"
        >
          <p>
            song2
          </p>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TrackList matches snapshot with non default selectedTrack 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Tracklist__Card-sc-1odr728-0 hULKon"
      >
        <div
          class="Tracklist__HeadingContainer-sc-1odr728-1 jjKLpr"
        >
          <span>
            <h2>
              Song
            </h2>
            <p>
              2
               Search Results
            </p>
          </span>
        </div>
        <ul
          class="Tracklist__TrackContainer-sc-1odr728-2 fvBEZw"
        >
          <li
            class="Tracklist___StyledLi-sc-1odr728-3 lnaxlV"
            style="opacity: 0; height: 0px;"
          >
            <p>
              song1
            </p>
          </li>
          <li
            class="Tracklist___StyledLi-sc-1odr728-3 cPRJbD"
            style="opacity: 0; height: 0px;"
          >
            <p>
              song2
            </p>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Tracklist__Card-sc-1odr728-0 hULKon"
    >
      <div
        class="Tracklist__HeadingContainer-sc-1odr728-1 jjKLpr"
      >
        <span>
          <h2>
            Song
          </h2>
          <p>
            2
             Search Results
          </p>
        </span>
      </div>
      <ul
        class="Tracklist__TrackContainer-sc-1odr728-2 fvBEZw"
      >
        <li
          class="Tracklist___StyledLi-sc-1odr728-3 lnaxlV"
          style="opacity: 0; height: 0px;"
        >
          <p>
            song1
          </p>
        </li>
        <li
          class="Tracklist___StyledLi-sc-1odr728-3 cPRJbD"
          style="opacity: 0; height: 0px;"
        >
          <p>
            song2
          </p>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
